contents/java/JavaReflections.md

```diff
@@ -177,105 +164,81 @@ An example implementation (not the actual) of the `Subscribe` annotation might b
 public @interface Subscribe { }
 ```
 
-And that is it! Important parts of the code to note are the first two lines before the declaration. The first line
-tells Java that this annotation must not be discarded during compile time so it will be available during runtime. The
-retention policy is there because some annotation do not mean anything after compilation (such as `Override` and
-`SuppressWarnings`), so it does not make sense to keep the annotation after compiling.
+And that is it! Important parts of the code to note are the first two lines before the declaration. The first line tells Java that this annotation must not be discarded during compile time so it will be available during runtime. The retention policy is there because some annotation do not mean anything after compilation (such as `Override` and `SuppressWarnings`), so it does not make sense to keep the annotation after compiling.
 
 The second line just means that this annotation can be applied to methods only.
 
-And the more important part is how the subscriber registry finds all its subscribing methods. The first step is to
-register a class as an event handler and an example of the code is like so:
+And the more important part is how the subscriber registry finds all its subscribing methods. The first step is to register a class as an event handler and an example of the code is like so:
 
 ```java
-
 public class Sheep extends Animal {
 
-    private static final EventsCenter EVENTS_BUS = EventsCenter.getInstance();
-
-    public final String name;
-
-    public Sheep(String nameOfSheep) {
-        name = nameOfSheep;
-        EVENTS_BUS.register(this);
-    }
-
-    @Subscribe
-    public void handleWeatherChangeEvent(WeatherChangeEvent event) {
-        if (event.weather == Weather.RAIN) {
-            hide();
-        }
+  private static final EventsCenter EVENTS_BUS = EventsCenter.getInstance();
+  
+  public Sheep() {
+    super();
+    EVENTS_BUS.register(this);
```

Comment:
I believe this should be `EVENTS_BUS.register(this.getClass())`, since the method asks for a `Class<?>` parameter and not `Object`.